cmd/skaffold/app/cmd/init.go

```diff
@@ -0,0 +1,312 @@
+		if err == nil {
+			logrus.Infof("found valid k8s yaml: %s", file)
+			k8sConfigs = append(k8sConfigs, file)
+			images = append(images, imgs...)
+		} else {
+			logrus.Infof("invalid k8s yaml %s: %s", file, err.Error())
+		}
+	}
+
+	var pairs []dockerfilePair
+	// conditionally generate build artifacts
+	if !skipBuild {
+		if len(dockerfiles) == 0 {
+			return errors.New("one or more valid Dockerfiles must be present to run skaffold; please provide at least one Dockerfile and try again")
+		}
+
+		if len(k8sConfigs) == 0 {
+			return errors.New("one or more valid kubernetes manifests is required to run skaffold")
+		}
+
+		if cliArtifacts != nil {
+			pairs, err = processCliArtifacts(cliArtifacts)
+			if err != nil {
+				return errors.Wrap(err, "processing cli artifacts")
+			}
+		} else {
+			pairs = resolveDockerfileImages(dockerfiles, images)
+		}
+	}
+
+	cfg, err := generateSkaffoldConfig(k8sConfigs, pairs)
+	if err != nil {
+		return err
+	}
+	if outfile != "" {
+		if err := ioutil.WriteFile(outfile, cfg, 0644); err != nil {
+			return errors.Wrap(err, "writing config to file")
+		}
+	} else {
+		out.Write(cfg)
+	}
+
+	return nil
+}
+
+func processCliArtifacts(artifacts []string) ([]dockerfilePair, error) {
+	var pairs []dockerfilePair
+	for _, artifact := range artifacts {
+		parts := strings.Split(artifact, "=")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("malformed artifact provided: %s", artifact)
+		}
+		pairs = append(pairs, dockerfilePair{
+			Dockerfile: parts[0],
+			ImageName:  parts[1],
+		})
+	}
+	return pairs, nil
+}
+
+// For each image parsed from all k8s manifests, prompt the user for
+// the dockerfile that builds the referenced image
+func resolveDockerfileImages(dockerfiles []string, images []string) []dockerfilePair {
+	// if we only have 1 image and 1 dockerfile, don't bother prompting
+	if len(images) == 1 && len(dockerfiles) == 1 {
+		return []dockerfilePair{{
+			Dockerfile: dockerfiles[0],
+			ImageName:  images[0],
+		}}
+	}
+	pairs := []dockerfilePair{}
+	for _, image := range images {
+		pair := promptUserForDockerfile(image, dockerfiles)
+		if pair.Dockerfile != NoDockerfile {
+			pairs = append(pairs, pair)
+			dockerfiles = util.RemoveFromSlice(dockerfiles, pair.Dockerfile)
+		}
+		images = util.RemoveFromSlice(images, pair.ImageName)
+	}
+	if len(dockerfiles) > 0 {
+		logrus.Warnf("unused dockerfiles found in repository: %v", dockerfiles)
+	}
+	return pairs
+}
+
+func promptUserForDockerfile(image string, dockerfiles []string) dockerfilePair {
+	var selectedDockerfile string
+	options := append(dockerfiles, NoDockerfile)
+	prompt := &survey.Select{
+		Message: fmt.Sprintf("Choose the dockerfile to build image %s", image),
+		Options: options,
+	}
+	survey.AskOne(prompt, &selectedDockerfile, nil)
+	return dockerfilePair{
+		Dockerfile: selectedDockerfile,
+		ImageName:  image,
+	}
+}
+
+func generateSkaffoldConfig(k8sConfigs []string, dockerfilePairs []dockerfilePair) ([]byte, error) {
```

Comment:
Can you add an integration test to make sure the config that we generate is actually a valid and runnable config? Something like

```
skaffold init
.. (input options)
skaffold run
```

Reply:
Yep, tests to come